Sources/SnapshotPreviewsCore/ExpandingViewController.swift

```diff
@@ -92,5 +118,42 @@ public final class ExpandingViewController: UIHostingController<EmergeModifierVi
     }
   }
 
+//  MARK: - Timer
+
+  func startTimer() {
+      if self.timer == nil {
+          self.startTime = Date()
+          self.timer = Timer.scheduledTimer(withTimeInterval: 0.1, repeats: true) { [weak self] _ in
+              guard let self else {
+                  return
+              }
+              if let start = self.startTime {
+                  self.elapsedTime = Date().timeIntervalSince(start)
+                  if self.elapsedTime >= HeightExpansionTimeLimitInSeconds {
+                      let scrollView = self.view.firstScrollView
+                      let timeoutError = RenderingError.expandingViewTimeout(CGSize(width: UIScreen.main.bounds.size.width, height: scrollView?.visibleContentHeight ?? -1))
+                      NSLog("ExpandingViewController: Expanding Scroll View timed out. Current height is \(scrollView?.visibleContentHeight ?? -1)")
+
+                      // Setting anchors back to full
+                      let fittingSize = self.sizeThatFits(in: UIScreen.main.bounds.size)
```

Comment:
I worry a bit about this sizeThatFits call only because it has to run on the main thread, and could take time/cause other layout changes. What is the reason for setting the anchors here rather than just calling runCallback?

Reply:
because in our discussion last week, I recall you saying if we time out, we should ensure that the size of the snapshot reverts back to "default" aka just the full screen

is that no longer the case?

Reply:
I don't think this is necessary/not sure why it would be helpful

Reply:
Even if we did need to change the size might make more sense to use the full screen like you said, but this would use a variable height right? Either way I don't think we should touch the heightAnchor once we decide it's going to be an error